### PR TITLE
Centralize environment dependent settings into environment package

### LINKED ISSUE #827: Centralize environment dependent settings into environment package
The `environment` package is already there to set settings based on environment settings and custom variable.

However, we still have some random location on the code where this should be moved.

https://github.com/threefoldtech/zos/blob/master/cmds/identityd/main.go#L234-L245
This should be set on environment package.

https://github.com/threefoldtech/zos/blob/master/pkg/flist/flist.go#L27
I think theses kind of settings should also be set on environment.

There is probably some other files too, I just saw theses two files by looking for something.

If we have all settings in the same place, it would be easier to know where to look to set custom settings, only change one place and later improve how to maybe custom that on runtime if needed.

### PR DESCRIPTION
**Pull request by DylanVerstraete**
Some other variables might be missing, @zaibon do you know where I missed some?

### CODE REVIEW

**Comment by zaibon** [comment]
From the top of my head like this I don't think of any other no.

File: pkg/environment/environment.go [hunk 1]
```diff
@@ -147,6 +156,10 @@ func getEnvironmentFromParams(params kernel.Params) (Environment, error) {
 		env.BcdbPassword = e
 	}
 
+	if e := os.Getenv("ZOS_FLIST_URL"); e != "" {
+		env.FlistURL = e
```
Comment (maxux) [inline pkg/environment/environment.go:151]: That's nice ! But if adding an environment variable for flist, could add one for BinRepo too ? :)

Comment (zaibon) [reply pkg/environment/environment.go:151]: Actually, how is this intended to be used ? There is no way to set a variable in a zinit service.

Comment (maxux) [reply pkg/environment/environment.go:151]: We have others environment variable aswell defined. In the past, we talked about a way to define environment variable easily for the whole system, that was never implemented but still could be. It's easier IMO than editing service file to add custom flags (like debug)

Comment (zaibon) [reply pkg/environment/environment.go:151]: alright, @maxux let's open a follow up issue  for this then.

**Pull request merged**